plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/CommonElementImporter.java

```diff
@@ -718,10 +723,91 @@ protected void parseParameter(final FBNetworkElement block) throws TypeImportExc
+		final var pathSegmenents = elements.subList(0, elements.size() - 1);
+		final var lastPathSegment = elements.get(elements.size() - 1);
+
+		final var vd = getVarConfigVD(typedSubApp, pathSegmenents, lastPathSegment);
+		if (vd != null) {
+			copyVarDeclFromType(vd, name, typedSubApp, pathSegmenents);
+		}
+	}
+
+	private static void copyVarDeclFromType(final VarDeclaration vd, final String path, FBNetworkElement typedSubApp,
+			final List<String> pathSegmenents) {
+		var fbn = typedSubApp.getFbNetwork();
+		final var parts = new ArrayList<>(List.of(path.split("\\."))); //$NON-NLS-1$
+
+		for (final var blockName : pathSegmenents) {
+			typedSubApp = fbn.getNetworkElements().stream().filter(elem -> elem.getName().equals(blockName)).findFirst()
+					.orElse(null);
+			if (typedSubApp == null) {
+				return;
+			}
+
+			if (typedSubApp instanceof final TypedSubApp tsa) {
+				fbn = tsa.getType().getFBNetwork();
+				if (!parts.isEmpty()) {
+					parts.remove(0);
+				}
+
+				final var relativeName = String.join(".", parts); //$NON-NLS-1$
+				final var copy = EcoreUtil.copy(vd);
+				copy.setName(relativeName);
+				copy.setValue(vd.getValue());
+				copy.setComment(vd.getComment());
+
+				if (!isAlreadyPresent(tsa.getVarConfigParams(), copy.getName())) {
+					tsa.getVarConfigParams().add(copy);
+				}
+			}
+		}
+	}
+
+	private static boolean isAlreadyPresent(final List<VarDeclaration> list, final String name) {
+		return list.stream().anyMatch(elem -> elem.getName().equals(name));
+	}
+
+	private static VarDeclaration getVarConfigVD(FBNetworkElement typedSubApp, final List<String> pathSegmenents,
+			final String lastPathSegment) {
+		var fbn = typedSubApp.getFbNetwork();
+		TypedSubApp current = null;
+
+		for (final var blockName : pathSegmenents) {
+			typedSubApp = fbn.getNetworkElements().stream().filter(elem -> elem.getName().equals(blockName)).findFirst()
+					.orElse(null);
+
+			if (typedSubApp == null) {
+				return null;
+			}
+
+			if (typedSubApp instanceof final TypedSubApp tsa) {
+				fbn = tsa.getType().getFBNetwork();
+				current = tsa;
+			} else if (typedSubApp instanceof final FB fb) {
+				final var vdOpt = fb.getInterface().getAllInterfaceElements().stream()
+						.filter(elem -> elem.getName().equals(lastPathSegment) && elem instanceof VarDeclaration)
+						.map(VarDeclaration.class::cast).findFirst();
+
+				if (vdOpt.isPresent()) {
+					final var vd = vdOpt.get();
+					if (current != null && current.getVarConfigParams().stream()
+							.noneMatch(param -> param.getName().equals(vd.getName()))) {
+						vd.setVarConfig(true);
+						return vd;
+					}
+				}
+			}
+		}
+
+		return null;
+	}
+
 	public static IInterfaceElement getInterfaceElement(final FBNetworkElement block, final String name,
 			final Value val) {
 		IInterfaceElement ie = block.getInterfaceElement(name);
-		if (null == ie) {
+		if (null == ie && !name.contains(".")) { //$NON-NLS-1$
```

Comment:
getInterfaceElement is used at several places inserting this change is for me to dangerouse.